### PR TITLE
Add calculation of alarm times

### PR DESCRIPTION
**Pull request by niccokunzmann**
This contributes to #716 by adding an `icalendar.Alarms` module.
This module allows calculating the alarm times.
It has been tested with

- RFC 9074
- Google
- Outlook
- Nextcloud
- Thunderbird
- Etar

TODO:

- [x] simple interface in Todo and Event
- [x] fix `# TODO:` comments
- [x] add changelog entry
- [x] implement tests that are todo
- [x] access relative alarm times from start and end as property of Alarms

----
📚 Documentation preview 📚: https://icalendar--737.org.readthedocs.build/

### CODE REVIEW

**Comment by niccokunzmann** [comment]
@stevepiercy, could you review the docs?
@jacadzaca, do you have time to have a look at this?

**Comment by stevepiercy** [comment]
@niccokunzmann FYI, I'm backlogged and slammed with preparation for [PloneConf 2024](https://2024.ploneconf.org/en). I estimate it will be at least a week until I can review icalendar PRs.

File: CHANGES.rst
Comment (stevepiercy) [inline CHANGES.rst]:
```suggestion
- Add ``Alarm.ACKNOWLEDGED``, ``Alarm.TRIGGER``, ``Alarm.REPEAT``, and ``Alarm.DURATION`` properties
```

File: src/icalendar/alarms.py
Comment (stevepiercy) [inline src/icalendar/alarms.py]:
```suggestion
        For example, in some calendar software, this is True until the user looks
```

File: src/icalendar/alarms.py
Comment (stevepiercy) [inline src/icalendar/alarms.py]:
```suggestion
        For example, Thunderbird and Google Calendar store the last time
```

File: src/icalendar/alarms.py
Comment (stevepiercy) [inline src/icalendar/alarms.py]:
```suggestion
        All alarms that happen before this time count as acknowledged.
```

File: src/icalendar/tools.py
Comment (stevepiercy) [inline src/icalendar/tools.py]:
```suggestion
    pytz requires this function to be used in order to correctly calculate the
```

File: src/icalendar/tests/test_pytz_zoneinfo_integration.py
Comment (stevepiercy) [inline src/icalendar/tests/test_pytz_zoneinfo_integration.py]:
```suggestion
    """Check that the provider name is OK."""
```

File: src/icalendar/tests/test_issue_716_alarm_time_computation.py
Comment (stevepiercy) [inline src/icalendar/tests/test_issue_716_alarm_time_computation.py]:
```suggestion
DTSTART, DTEND, and DURATION.
```

File: src/icalendar/cal.py
Comment (stevepiercy) [inline src/icalendar/cal.py]:
```suggestion
    When an alarm is triggered on a client, clients can check to see if an "ACKNOWLEDGED"
```

File: src/icalendar/cal.py
Comment (stevepiercy) [inline src/icalendar/cal.py]:
```suggestion
        RDATE, EXDATE, and RRULE properties.
```

File: src/icalendar/cal.py
Comment (stevepiercy) [inline src/icalendar/cal.py]:
```suggestion
        RDATE, EXDATE, and RRULE properties.
```

File: src/icalendar/cal.py
Comment (stevepiercy) [inline src/icalendar/cal.py]:
```suggestion
        Note that this only uses DTSTART and DTEND, but ignores
```

File: src/icalendar/cal.py
Comment (stevepiercy) [inline src/icalendar/cal.py]:
```suggestion
        Note that this only uses DTSTART and DUE, but ignores
```

File: src/icalendar/cal.py
Comment (stevepiercy) [inline src/icalendar/cal.py]:
```suggestion
        """Whether this component has attributes that indicate that Mozilla Thunderbird created it."""
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I caught only a few typos and Oxford comma issues. This is an excellent feature.

I approve once you review and make the changes.

**Comment by niccokunzmann** [comment]
@stevepiercy Thanks for reviewing the docs!

After I merging this, I will use the functionality for recurring-ical-events to calculate if an alarm happens within a time span.
I might come back with improvements in follow-up PRs.

**Comment by niccokunzmann** [comment]
Thanks for the review again! The tests pass and I added all your suggestions. Merging ...

**Pull request merged**
